src/CommandProcessor.php

```diff
@@ -137,8 +141,19 @@ public function validateRunAndAlter(
             return $validated;
         }
 
-        // Run the command, alter the results, and then handle output and status
-        $result = $this->runCommandCallback($commandCallback, $commandData);
+        $replaceDispatcher = new ReplaceCommandHookDispatcher($this->hookManager(), $names);
+        if ($this->logger) {
```

Comment:
That will not work since the `setLogger()` method type hints its first argument with `LoggerInterface`.